testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -493,9 +358,325 @@ def test_no_units(h5md_file, ref, tmpdir):
+        chunks = (1, 1000, 3)
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    chunks=chunks) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        dset = uw.trajectory._particle_group['position/value']
+        assert_equal(dset.chunks, chunks)
+
+    @pytest.mark.parametrize('filter, opts', (('gzip', 1),
+                                              ('gzip', 9),
+                                              ('lzf', None)))
+    def test_write_with_compression(self, universe,
+                                    outfile, Writer,
+                                    filter, opts):
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    compression=filter,
+                    compression_opts=opts) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        dset = uw.trajectory._particle_group['position/value']
+        assert_equal(dset.compression, filter)
+        assert_equal(dset.compression_opts, opts)
+
+    @pytest.mark.parametrize('driver, comm', (('core', None),
+                                              ('mpio', 'comm')))
+    def test_write_with_drivers(self, universe,
+                                outfile, Writer,
+                                driver, comm):
+        try:
+            with Writer(outfile,
+                        universe.atoms.n_atoms,
+                        driver=driver,
+                        comm=comm) as W:
+                for ts in universe.trajectory:
+                    W.write(universe)
+
+            uw = mda.Universe(TPR_xvf, outfile, driver=driver, comm=comm)
+            file = uw.trajectory._file
+            assert_equal(file.driver, driver)
+        except:
+            with pytest.raises(ValueError):
+                with Writer(outfile,
+                            universe.atoms.n_atoms,
+                            driver=driver,
+                            comm=comm) as W:
+                    for ts in universe.trajectory:
+                        W.write(universe)
+
+    def test_timestep_not_modified_by_writer(self, universe, Writer, outfile):
+        trj = universe.trajectory
+        ts = trj.ts
+
+        trj[-1]  # last timestep (so that time != 0)
+        x = ts._pos.copy()
+        time = ts.time
+
+        with Writer(outfile, trj.n_atoms) as W:
+            # last timestep (so that time != 0) (say it again, just in case...)
+            trj[-1]
+            W.write(universe)
+
+        assert_equal(
+            ts._pos,
+            x,
+            err_msg="Positions in Timestep were modified by writer.")
+        assert_equal(
+            ts.time, time, err_msg="Time in Timestep was modified by writer.")
+
+    """def test_gaps(self, universe, Writer, outfile):
```

Comment:
You can mark tests as XFAIL if that's a known problem. It serves as documentation and a good way to motivate future work — all one has to do is to make the XFAILS pass!